### PR TITLE
mgr/orch: add --all-available-devices to 'orch apply osd'

### PR DESCRIPTION
**Pull request by liewegas**
Provide a super-simple "use any avaialable device" command for
'orch apply osd'.  This will work for many (maybe even most?) users.

Signed-off-by: Sage Weil <sage@redhat.com>

### CODE REVIEW

File: src/pybind/mgr/orchestrator/module.py [hunk 1]
```diff
@@ -429,16 +429,28 @@ def ukn(s):
...
+            if inbuf:
+                raise OrchestratorError('--all-available-devices cannot be combined iwth a osd spec')
+            inbuf = """
+service_type: osd
+service_id: all-available-devices
+placement:
+  host_pattern: '*'
+data_devices:
+  all: True
+"""
```
Comment (sebastian-philipp) [inline src/pybind/mgr/orchestrator/module.py]: please make this statically verifiable:
```suggestion
            dg_spec = DriveGroupSpec(service_id='all-available-devices',
                                                         placement=PlacementSpec(host_pattern='*')
                                                         data_devices=DeviceSelection(all=True)
                                                         )
```

**Comment by liewegas** [comment]
retest this please

File: src/pybind/mgr/orchestrator/module.py [hunk 2]
```diff
@@ -429,22 +429,34 @@ def ukn(s):
...
             return HandleCommandResult(-errno.EINVAL, stderr=usage)
-        try:
-            drivegroups = yaml.load_all(inbuf)
-            dg_specs = [ServiceSpec.from_json(dg) for dg in drivegroups]
-        except ValueError as e:
-            msg = 'Failed to read JSON/YAML input: {}'.format(str(e)) + usage
-            return HandleCommandResult(-errno.EINVAL, stderr=msg)
+        if all_available_devices:
+            if inbuf:
+                raise OrchestratorError('--all-available-devices cannot be combined iwth a osd spec')
```
Comment (sebastian-philipp) [inline src/pybind/mgr/orchestrator/module.py]:
```suggestion
                raise OrchestratorError('--all-available-devices cannot be combined with a osd spec')
```

File: src/pybind/mgr/orchestrator/module.py [hunk 3]
```diff
@@ -429,22 +429,34 @@ def ukn(s):
...
-    def _apply_osd(self, inbuf=None):
-        # type: (Optional[str]) -> HandleCommandResult
+        'name=all_available_devices,type=CephBool,req=false',
+        'Create OSD daemon(s) using a drive group spec')
+    def _apply_osd(self, all_available_devices=False, inbuf=None):
+        # type: (bool, Optional[str]) -> HandleCommandResult
         """Apply DriveGroupSpecs to create OSDs"""
         usage = """
 Usage:
   ceph orch apply osd -i <json_file/yaml_file>
```
Comment (sebastian-philipp) [inline src/pybind/mgr/orchestrator/module.py]:
```suggestion
  ceph orch apply osd -i <json_file/yaml_file>
  ceph orch apply osd  --all_available_devices
```

File: src/pybind/mgr/orchestrator/module.py [hunk 4]
```diff
@@ -429,22 +429,34 @@ def ukn(s):
 
     @_cli_write_command(
         'orch apply osd',
-        desc='Create an OSD daemons using drive_groups')
-    def _apply_osd(self, inbuf=None):
-        # type: (Optional[str]) -> HandleCommandResult
+        'name=all_available_devices,type=CephBool,req=false',
```
Comment (sebastian-philipp) [inline src/pybind/mgr/orchestrator/module.py]:
```suggestion
        'name=all-available-devices,type=CephBool,req=false',
```

Comment (liewegas) [reply src/pybind/mgr/orchestrator/module.py]: param names always have _ not -

**Pull request merged**